Framework/Algorithms/src/CalculateTransmissionBeamSpreader.cpp

```diff
@@ -244,7 +245,7 @@ API::MatrixWorkspace_sptr
 CalculateTransmissionBeamSpreader::extractSpectrum(API::MatrixWorkspace_sptr WS,
                                                    const size_t index) {
   // Check that given spectra are monitors
-  if (!WS->getDetector(index)->isMonitor()) {
+  if (!WS->spectrumInfo().isMonitor(index)) {
```

Comment:
How often is this method called? Currently `MatrixWorkspace::spectrumInfo()` is a bit expensive, so doing so for every `index` might need to be avoided.